src/scripting_api/event.js

```diff
@@ -47,11 +47,32 @@ class EventDispatcher {
     this._document.obj._eventDispatcher = this;
   }
 
+  /*
+   * Take an event object and reconstruct what we think the result will be once
+   * the change is applied, so we can validate against it and cancel the event
+   * if need be
```

Comment:
Nit: Here, and everywhere in the patch, please make sure to always end all sentences with a period.

Also, please keep https://github.com/mozilla/pdf.js/wiki/Squashing-Commits in mind when updating the patch.